### DISCUSSION TITLE
A B2 relationship that creates an alias field for the other side of a M2O relationship

### DISCUSSION THREAD

**Discussion by emahuni** [directus/directus#19584] [Feature Requests]
### Summary

Directus is missing a `Belongs To` relationship that allows resolution of the `M2O `relationship without creating an additional foreign key field in the related collection. This should be easy to do since the logic already exists with `M2O < - > O2M` relationships pair. The end result should be `has_one` (M2O field, locked with unique constraint) <-> `belongs_to` (alias that does what `O2M` does but for single item).

The O2O example given in docs is just a has one (One Way) relationship where there is no aliased One to One relationship such as the M2O <-> O2M relationships do.

Once again, this is an excellent framework that's on its own league and I love it.

### Basic Example

I have  a `user` that has a `profile`, or to put it the other way round, a `profile` that belongs to a `user`. So it's `user <-  profile` relationship that I want to maintain. If the `user` is deleted, or if the `user.profile` side is changed to a new `profile` or set to `null`, the old `profile.user` side should resolve to nothing since the `profile` now **belongs to** nothing or we can delete the `profile` altogether (relational triggers), or if the `profile` is deleted, or if the `profile.user` side is set to a new `user` or another `user`, the old `user.profile` side is set to `null` or deleted (relational triggers), since the `user` doesn't have a `profile` anymore.

### Motivation

The M2O - O2M relationships pair is just exactly how I expect this kind of relationship to work, where one can optionally instruct Directus to create a `Belongs 2` field on the opposite side of the `Many 2 One` relationship. When the B2 relationship is created, it constraints the M2O relationship to a unique ID.

### Detailed Design

Similar to how M2O - O2M relationship works.

### Requirements List

_

### Drawbacks

__

### Alternatives

I could create a custom extension that uses hooks to do this on update, create, delete, read etc. but this will be slower compared to this being implemented deep into the core like all other relationships. This will end up running a fresh knex query to join and append to the request.

### Adoption Strategy

__

### Unresolved Questions

not sure if I have access to all the data needed to correctly filter and select the appropriate fields for the custom extension.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
